Components/ImportService.php

```diff
@@ -332,16 +332,18 @@ public function storeRemoteCategories(array $remoteItems)
 
         $connection->beginTransaction();
         try {
-            foreach ($remoteItems as $articleId => $categories) {
+            foreach ($remoteItems as $articleId => $remoteItem) {
+                $categories = $remoteItem[0];
+                $shopId = $remoteItem[1];
```

Comment:
maybe it's time to create an object with following properties:
* articleId
* categories array
* shopId

Actually you can use SDK Product. ```$remoteItems[$articleId] = new Product```.

It's hard to work with arrays and indexes. Always need to check if the index exists.

Reply:
this functions is completely duplicated by CategoryResolver::storeRemoteCategories
in categoryResolver it has proper input parameter